hiding-in-the-clouds_secret-management-with-openstack-barbican/heat-templates/master/cinder-encrypted.yaml

```diff
@@ -0,0 +1,166 @@
+# Example invocation: heat stack-create --poll -f cinder-encrypted.yaml cinder-encrypted
+heat_template_version: 2015-10-15
+
+# Some notes about the parameters:
+#
+# The parameters provided are the defaults for Openstack SUSE.
+# For devstack, the following values apply:
+#
+# floating_network: public
+# image: cirros-0.3.4-x86_64-uec
+#
+parameters:
+  floating_network:
+    type: string
+    default: floating
+  image:
+    type: string
+    default: cirros-0.3.4
+  flavor:
+    type: string
+    default: m1.tiny
+  key_name:
+    type: string
+    default: mykey
+    description: |
+      This keypair needs to exist in Nova. Create as follows (assuming your SSH
+      public key is in ~/.ssh/id_rsa.pub):
+
+      nova keypair-add --pub-key ~/.ssh/id_rsa.pub mykey
+
+
+resources:
+
+  mynetwork:
+    type: OS::Neutron::Net
+    properties:
+      name: mynet
+
+
+  mysubnet:
+    type: OS::Neutron::Subnet
+    properties:
+      cidr: 10.0.0.1/24
+      name: mysubnet
+      network:
+        get_resource: mynetwork
+
+  # Note: this can only created if heat-engine uses an adminURL for its Cinder client
+  # cinder type-create encrypted
+  myvolumetype:
+    type: OS::Cinder::VolumeType
+    properties:
+      name: encrypted
+      is_public: true
+
+  # Note: this can only created if heat-engine uses an adminURL for its Cinder client
+  # cinder encryption-type-create encrypted nova.volume.encryptors.luks.LuksEncryptor
+  #
+  # Note: in mitaka, the cipher and key_size were optional and defaulted to the values
+  # seen below.  In newton, it appears that these defaults have been removed and
+  # these values must be added.
```

Comment:
Nitpick: s/values/properties/ (Heat refers to parameters that go into a resource as "properties" and to parameters that are retrieved from an existing resource as "attributes")